src/azul/indexer/__init__.py

```diff
@@ -376,9 +375,24 @@ def parse(cls, spec: str) -> Self:
         raise NotImplementedError
 
     @classmethod
-    def _parse(cls, spec: str) -> tuple[str, Prefix | None]:
+    def extract_prefix(cls, spec: str) -> tuple[str, Prefix | None]:
+        """
+        Extract and parse the prefix from a string representation of a
+        `SourceSpec`. To parse the entire spec, use :meth:`parse`.
```

Comment:
This doesn't explain the return value. Since the only public usage of this method discards the first element of the returned tuple, we can simplify the signature by 1) leaving this method as `_parse()` and 2) adding a public (and documented) `parse_prefix_only()` wrapper that just returns the prefix. Simplifying the signature eliminates the need to explain the tuple elements. For `parse_prefix_only` it should be document what a `None` return value means.